cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go

```diff
@@ -71,8 +71,10 @@ type MasterConfiguration struct {
 
 // API struct contains elements of API server address.
 type API struct {
-	// AdvertiseAddress sets the address for the API server to advertise.
+	// AdvertiseAddress sets the IP address for the API server to advertise.
 	AdvertiseAddress string `json:"advertiseAddress"`
+	// AdvertiseDNSAddress sets the DNS address for the API server to advertise.
+	AdvertiseDNSAddress string `json:"advertiseDNSAddress"`
```

Comment:
I find this naming a bit confusing as I don't think anything is being "advertised."  While the `kubelet` uses the terminology of `advertiseAddress` to specify which IP the server is listening on, this new argument is being used only to indicate where a node should register itself.  Perhaps a more descriptive argument would be `controlPlaneEndpoint`?

Reply:
I think controlPlane make sense, I'll move to that, take out the args, and do a verify it works via the config.